---
title: Apps.getSelf
slug: Archive/Marketplace/API/DOMApplicationsRegistry/getSelf
tags:
  - API Apps
  - Apps
translation_of: Archive/Marketplace/API/DOMApplicationsRegistry/getSelf
---
<p>{{ ApiRef() }}</p>
<p>{{ non-standard_header() }}</p>
<h2 id="Résumé">Résumé</h2>
<p>Renvoie les informations sur l'application courante, ce qui veut dire une application installée dont le domaine correspond au domaine de l'application appelée.</p>
<div class="note">
 <p><strong>Note :</strong> Plusieurs applications pour une même origine ne sont pas supportées. Pour héberger plusieurs applications depuis un domaine, <a href="/en-US/docs/Apps/Adding_a_subdomain" title="/en-US/docs/Apps/Adding_a_subdomain">créer un sous-domaine pour chaque application</a>. Par exemple : <code>monapp.mondomaine.com</code>, <code>autreapp.mondomaine.com</code>, et ainsi de suite.</p>
</div>
<h2 id="Syntaxe">Syntaxe</h2>
<pre><code>var requete = window.navigator.mozApps.getSelf()</code>;
</pre>
<h2 id="Paramètres">Paramètres</h2>
<p>Aucun.</p>
<h2 id="Renvoie">Renvoie</h2>
<p><code>getSelf()</code> renvoie un objet {{ domxref("DOMRequest") }}. Le champ <code>DOMRequest.result</code> contient un <a href="/fr/docs/DOM/App">objet <code>App</code></a>, qui est un objet JavaScript décrivant l'application. Avant que l'opération soit terminée, <code>DOMRequest.result</code> est <code>null</code>.</p>
<div class="note">
 <p><strong>Note :</strong> A cause du {{ Bug("806597") }}, <code>requete.result</code> retourne incorrectement <code>null</code> même sur l'application tourne sur l'ordinateur ou Firefox pour Android.</p>
</div>
<p>Si l'appel est un echec, <code>DOMRequest.error </code>contient un objet {{ domxref("DOMError") }}, qui contient des informations sur l'erreur.</p>
<h2 id="Exemple">Exemple</h2>
<p>Un exemple montrant comme utiliser <code>getSelf()</code> avec les propriétés de callback <code>DOMRequest.onsuccess</code> et <code>DOMRequest.onerror.</code></p>
<pre class="brush: js">var requete = window.navigator.mozApps.getSelf();
requete.onsuccess = function() {
  // Affiche le nom de l'application depuis l'objet App
  alert("Nom de l'application : " + requete.result.manifest.name);
};
requete.onerror = function() {
  // Affiche le nom de l'erreur depuis l'objet DOMError
  alert("Erreur : " + requete.error.name);
};</pre>
<p>Si l'appel est réussi, un objet {{ domxref("App") }} est renvoyé dans la propriété <code>result</code> de l'objet renvoyé. Dans cette exemple, il s'agit de <code>requete.result</code>.</p>
<h2 id="Sujets_en_relation">Sujets en relation</h2>
<ul>
 <li><a href="/fr/docs/Applications/API_JavaScript_Apps">API JavaScript Apps</a></li>
</ul>
